js/views/messagesitem.js

```diff
@@ -126,42 +146,79 @@ define(function(require) {
 			// directly change star state in the interface for quick feedback
 			if (starred) {
 				this.getUI('star')
-						.removeClass('icon-starred')
-						.addClass('icon-star');
+					.removeClass('icon-starred')
+					.addClass('icon-star');
 			} else {
 				this.getUI('star')
-						.removeClass('icon-star')
-						.addClass('icon-starred');
+					.removeClass('icon-star')
+					.addClass('icon-starred');
 			}
 
 			Radio.message.trigger('flag', this.model, 'flagged', !starred);
 		},
+
 		openMessage: function(event) {
+			// Ignore the event if the actions menu was targeted
+			if ($(event.target).hasClass('toggle-menu')) {
+				return;
+			}
 			event.stopPropagation();
 			$('#mail-message').removeClass('hidden-mobile');
 			// make sure message is marked as read when clicked on it
 			Radio.message.trigger('flag', this.model, 'unseen', false);
-			Radio.message.trigger('load', this.model.folder.account, this.model.folder, this.model, {
-				force: true
-			});
+			Radio.message.trigger(
+				'load',
+				this.model.folder.account,
+				this.model.folder,
+				this.model,
+				{
+					force: true
+				}
+			);
 		},
-		deleteMessage: function(event) {
+
+		/**
+		 * @param {Event} e
+		 */
+		toggleActionsMenu: function(e) {
+			e.preventDefault();
+			this.actionsMenuShown = !this.actionsMenuShown;
+			this.toggleMenuClass();
+		},
+
+		toggleMenuClass: function() {
+			this.getUI('menu').toggleClass('open', this.actionsMenuShown);
+		},
+
+		toggleMessageRead: function(event) {
```

Comment:
- [ ] BUG: after clicking `Toggle read`, the menu is neither closed, nor can I close it by clicking the three dots again. IMO I'd go with closing the menu right away after one of the action inside has been clicked.

Reply:
@ChristophWurst  I unfortunately cannot replicate this on Chrome 67 and Firefox 61
On my end, the menu is closed immediately the `Toggle read` is clicked.

Reply:
Oh, wait. I think I described the issue wrong. The issue is actually that you can't re-open the menu after one of the actions has been clicked (I guess the toggle flag is missing an update). You have to click the three dots twice to get back into the menu.

Reply:
Ahh, now I can reproduce the bug :)
I'll take a look at it.
Thanks for the better explanation 👍

Reply:
@ChristophWurst the bug should be fixed now.